File: app/scrapers/interfaces/movie.py
from abc import ABC, abstractmethod
from dataclasses import dataclass


class MovieProvider(ABC):
    @abstractmethod
    def getMovieCollection(self):
        raise NotImplementedError()

    @abstractmethod
    def searchMovie(self, name, year=-1):
        raise NotImplementedError()

    @abstractmethod
    def getMoviePeople(self):
        raise NotImplementedError()

    @abstractmethod
    def getMovieTags(self):
        raise NotImplementedError()

    @abstractmethod
    def getMovie(self):
        raise NotImplementedError()


@dataclass(frozen=True)
class MovieData:
    title: str
    overview: str
    icon: str
    fanart: str
    rating: float
    premiered: str
    collection: int
    scraperID: str
    scraperName: str
    scraperData: str
    scraperLink: str


@dataclass(frozen=True)
class MovieCollectionData:
    title: str
    overview: str
    icon: str
    fanart: str
    premiered: str
    scraperID: str
    scraperName: str
    scraperData: str
    scraperLink: str

File: internal/scraper/common/movie.go
package common

type MovieProvider interface {
	Provider
	SearchMovie(name string, year int) ([]SearchData, error)
	ListMovieTag() ([]TagData, error)
	ListMoviePerson() ([]PersonData, error)
	GetMovie() (MovieData, error)
	GetMovieUpcoming() (UpcomingData, error)
	GetMovieCollection() (MovieCollectionData, error)
}

type MovieData struct {
	Title      string `json:"title"`
	Overview   string `json:"overview"`
	Icon       string `json:"icon"`
	Fanart     string `json:"fanart"`
	Website    string `json:"website"`
	Trailer    string `json:"trailer"`
	Premiered  int64  `json:"premiered"`
	Rating     int64  `json:"rating"`
	Collection int64  `json:"collection"`
	ScraperInfo
}

type MovieCollectionData struct {
	Title     string `json:"title"`
	Overview  string `json:"overview"`
	Icon      string `json:"icon"`
	Fanart    string `json:"fanart"`
	Premiered int64  `json:"premiered"`
	Rating    int64  `json:"rating"`
	ScraperInfo
}
